docs/src/man/blockarrays.md

```diff
@@ -6,13 +6,28 @@ DocTestSetup = quote
 end
 ```
 
+
+## Creating initialized `BlockArrays`
+
+A block array can be created with initialized blocks using the `BlockArray{T}(block_sizes)`
+function. The block_sizes are each an `AbstractVector{Int}` which determines the size of the blocks in that dimension. We here create a `[1,2]×[3,2]` block matrix of `Float32`s:
```

Comment:
Perhaps we should mention that the arrays themselves are initialized, the values in them are not? What do you think?

Reply:
Is this behaviour going to change in 0.7 so that `Matrix{Float64}(n,m)` automatically sets the values?

Reply:
I don't think it will, but not sure.

Reply:
I'm a bit confused about the difference between `Matrix{Float64}(uninitialized, n, m)` and `Matrix{Float64}(n, m)`.  

I'm worried that there's a confusion between uninitialized blocks and uninitialized entries with using `uninitialized` to mean  uninitialized blocks. Especially since I'm hoping to making `PseudoBlockArray` and `BlockArray` have equivalent behaviour for the same syntax. 

It might be better to add a `UninitializedBlocks` and then we have the following three cases:
1. `BlockArray{Float64}(uninitialized_blocks, ns)`, which does not initialize the blocks, i.e., `A.blocks[1]` is `#undef`. This is the current behaviour of `BlockArray(Vector{Float64}, ns, ms)`
2. `BlockArray{Float64}(uninitialized, ns)`, which initializes the blocks, but not the entries of the blocks, i.e., `A[Block(1)]` is allocated as a `Vector{Float64}(uninitialized, ns[1])`
3. BlockArray{Float64}(ns), which does initialize the blocks, and the entries, i.e., `A[Block(1)]` is allocated as a `Vector{Float64}(ns[1])`
2. `PseudoBlockArray{Float64}(uninitialized, ns)`,  `A. blocks` is allocated as a `Vector{Float64}(uninitialized, sum(ns))`
3. BlockArray{Float64}(ns), which does initialize the blocks, and the entries, i.e., `A.blocks` is allocated as a `Vector{Float64}(sum(ns))`